web/src/server_events_dispatch.js

```diff
@@ -721,19 +721,37 @@ export function dispatch_normal_event(event) {
                 // present in the backend/Jinja2 templates.
                 settings_display.set_default_language_name(event.language_name);
             }
-            if (
-                event.property === "default_view" && // If current hash is empty (default view), and the
+            if (event.property === "default_view") {
+                const new_default_view = event.value;
+                const $recent_view_sidebar_menu_icon = $(".recent-view-sidebar-menu-icon");
+                const $all_messages_sidebar_menu_icon = $(".all-messages-sidebar-menu-icon");
+                if (new_default_view === settings_config.default_view_values.all_messages.code) {
+                    $recent_view_sidebar_menu_icon.removeClass("hide");
+                    $all_messages_sidebar_menu_icon.addClass("hide");
+                } else if (
+                    new_default_view === settings_config.default_view_values.recent_topics.code
+                ) {
+                    $recent_view_sidebar_menu_icon.addClass("hide");
+                    $all_messages_sidebar_menu_icon.removeClass("hide");
+                } else {
+                    // Inbox is default view.
+                    $recent_view_sidebar_menu_icon.removeClass("hide");
+                    $all_messages_sidebar_menu_icon.removeClass("hide");
+                }
```

Comment:
Let's put this code in a function like `left_sidebar_navigation_area.handle_home_view_change`  -- we generally don't put jQuery code in this file and that also helps localize the code in the module containing related logic.

Reply:
Done.